pkg/admission/team_webhook.go

```diff
@@ -89,3 +95,16 @@ func validateGreenhouseLabels(team *greenhousev1alpha1.Team, ctx context.Context
 	}
 	return nil
 }
+
+func validateJoinURL(team *greenhousev1alpha1.Team) error {
+	if team.Spec.JoinURL == "" {
+		return nil
+	}
+	if !validateURL(team.Spec.JoinURL) {
+		return apierrors.NewInvalid(team.GroupVersionKind().GroupKind(), team.GetName(), field.ErrorList{
+			field.Invalid(field.NewPath("spec").Child("joinUrl"), team.Spec.JoinURL,
+				"JoinURL must be a valid URL."),
```

Comment:
The error message should include what the webhook expects from a valid URL. This way the user knows what to do when they see the error.